code-challenges.test.js

```diff
@@ -13,34 +15,69 @@
+}
 
 // --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
 
 // a) Create a test with an expect statement using the variables provided.
-
-const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
-// Expected output: [ 2, 0, -1, 0 ]
-const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
-// Expected output: [ 2, 1, -1 ]
+describe('onlyRemainders', () => {
+  it('Takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.', () => {
+    const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+    // Expected output: [ 2, 0, -1, 0 ]
+    const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
+    // Expected output: [ 2, 1, -1 ]
+    expect(onlyRemainders(hodgepodge1)).toEqual([2, 0, -1, 0])
+    expect(onlyRemainders(hodgepodge2)).toEqual([2, 1, -1])
+  })
+})
 
 // b) Create the function that makes the test pass.
+// To create this function, I will first take in an array as an argument, and then filter out any values whose typeof does not return "number". Then, I will map over this filtered array, using the modulo operator to find the remainder of eachvalue divided by 3. I will return the new mapped array.
+const onlyRemainders = (array) => {
+  return array.filter((value) => typeof value === 'number').map((num) => num % 3)
+}
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
-
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
+describe('sumOfCubes', () => {
+  it('Takes in an array of numbers and returns the sum of all the numbers cubed.', () => {
+    const cubeAndSum1 = [2, 3, 4]
+    // Expected output: 99
+    const cubeAndSum2 = [0, 5, 10]
+    // Expected output: 1125
+    expect(sumOfCubes(cubeAndSum1)).toEqual(99)
+    expect(sumOfCubes(cubeAndSum2)).toEqual(1125)
+  })
+})
 
 // b) Create the function that makes the test pass.
+// To create this function, I will first take in an array as an argument. Then I will create a variable called sum. I will use foreach on the array to iterate through and cube each value, adding it to sum after. Then I will return sum.
+
+// const sumOfCubes = (array) => {
+//   let sum = 0
+//   array.forEach((num) => sum += (num ** 3));
+//   return sum
+// }
+
+
+// After some research, I was able to refactor my code using the .reduce method. This method will initialize a 'beginning' variable, which I call sum. It initializes to a given value, where I chose 0. Then, it will iterate over the array performing the defined function for each value(I called num) and return sum, the beginning variable.
```

Comment:
Great research and explanation of `.reduce`! Shows that you took the time to understand what was happening before using it!